test/WebJobs.Script.Tests/UtilityTests.cs

```diff
@@ -595,5 +600,74 @@ public void TryGetFunctionName_ReturnsExpectedResult(string keyName, bool expect
                 Assert.Equal("test", functionName);
             }
         }
+
+        [Fact]
+        public void Test_LogAutorestGeneratedJson_Without_AutorestGeneratedJson()
+        {
+            string GetTemporaryDirectory()
+            {
+                string tempDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+                Directory.CreateDirectory(tempDirectory);
+                return tempDirectory;
+            }
+
+            string functionAppDirectory = GetTemporaryDirectory();
+            Utility.LogAutorestGeneratedJson(functionAppDirectory, _testLogger);
+            var allLogs = _testLogger.GetLogMessages();
+            Assert.Empty(allLogs);
+        }
+
+        [Fact]
+        public void Test_LogAutorestGeneratedJson_With_Valid_AutorestGeneratedJson()
+        {
+            string GetTemporaryDirectory()
+            {
+                string tempDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+                Directory.CreateDirectory(tempDirectory);
+                return tempDirectory;
+            }
+
+            string functionAppDirectory = GetTemporaryDirectory();
+            string sample_autorest_generated_json = @"{
+  ""name"": ""@autorest/azure-functions-csharp"",
+  ""version"": ""0.2.0-preview""
+}
+";
+            string expected_contents = @"autorest_generated.json file found generated by Autorest (https://aka.ms/stencil) | file content";
+            File.WriteAllText(Path.Combine(functionAppDirectory, ScriptConstants.AutorestGeenratedMetadataFileName), sample_autorest_generated_json);
+            Utility.LogAutorestGeneratedJson(functionAppDirectory, _testLogger);
+            var allLogs = _testLogger.GetLogMessages();
+            VerifyLogLevel(allLogs, expected_contents, LogLevel.Information);
+        }
+
```

Comment:
combine all these tests using `[Theory]` taking input context and expect log as input.